### PR TITLE
Feature run multi region case as single case

### PR DESCRIPTION
**Pull request by denMaier**
When a case is set up for a multi region calculation, this feature lets one decide to run only a single region at a time. 

This might be useful to test the convergence of that region, or find mistakes in the case setup.

### CODE REVIEW

**Comment by philipcardiff** [comment]
Hi @denMaier, this is an interesting concept. Do you have a test case that demonstrates this?
Or, let me know how I can try this out on an existing tutorial.

**Comment by iBatistic** [comment]
Hi @philipcardiff , @denMaier 

The check_added_files script is failing on solidModel.C because he has trailing spaces at the end of two lines. This script reads the files before and after calling perl command to replace the file banner. Perl's command didn't do anything; however, these trailing spaces are lost and because of that, differences exist and the script fails. I will think about how to modify this script, so we can know when we have trailing spaces. In the meantime, I suggest leaving it as it is, and we can use it to detect trailing spaces.

**Comment by denMaier** [comment]
Hi @philipcardiff,  its not an awefully interesting case when uncoupled, but you could use beamInCrossFlow. With the code changes proposed, you just need add a subdict 
```
solid
{
  region solid;
} 
```
to the 'controlDict' and change the physicsModel to 'solid'.
In this case, you might only be able to see if the calculation starts as expected, but you could also add a force to the boundary to check convergence properties. 
The advantage is, that you don't have to copy all the files into a new case with the right folder structure (no 'solid' subfolders) to check the solid case alone.

**Comment by denMaier** [comment]
hey @iBatistic ,  is there anything i need to do?  How can i find the lines with trailing spaces?

**Comment by iBatistic** [comment]
Hi @denMaier , In some text editors you can set trailing spaces to be colored, so you can detect them. Also, for example in Vim and Emacs, you have a specific keystroke to delete them.

I also often forget to delete them when pushing code, I guess this coding style rule is easier to break in OpenFoam xD.

File: src/solids4FoamModels/solidModels/solidModel/solidModel.C [hunk 1]
```diff
@@ -1612,16 +1612,23 @@ Foam::autoPtr<Foam::solidModel> Foam::solidModel::New
     const word& region
 )
 {
+
+    // In case a case, that is set up for a muti-region calculation, is supposed to be run as simple (single region)
+    // calculation (i. e. to check the convergence of that single region) the solver can be
+    // set to solid in physicsProperties and in the controlDict under the subDict 'solid' one can select the region.
+    // This also allows the region name of a calulation not to be 'region0' or 'solid' but user defined.
+    word runRegion(runTime.controlDict().subOrEmptyDict("solid").lookupOrDefault<word>("region",region));
```
Comment (philipcardiff) [inline src/solids4FoamModels/solidModels/solidModel/solidModel.C:1620]: Thanks @denMaier, please add a space ` ` between `"region",` and `region` and then I will merge this.

**Pull request merged**